### ISSUE TITLE
Influxdata Telegraf inputs.logparser

### ISSUE THREAD

**Issue by gbodelle** [influxdata/telegraf#7887]
### Relevant telegraf.conf:
```
# Stream and parse log file(s).
[[inputs.logparser]]
  ## Log files to parse.
  ## These accept standard unix glob matching rules, but with the addition of
  ## ** as a "super asterisk". ie:
  ##   /var/log/**.log     -> recursively find all .log files in /var/log
  ##   /var/log/*/*.log    -> find all .log files with a parent dir in /var/log
  ##   /var/log/apache.log -> only tail the apache log file
  files = ["/etc/telegraf/telegraf.d/monitor.log"]

  ## Read files that currently exist from the beginning. Files that are created
  ## while telegraf is running (and that match the "files" globs) will always
  ## be read from the beginning.
  from_beginning = true

  ## Method used to watch for file updates.  Can be either "inotify" or "poll".
  # watch_method = "inotify"

  ## Parse logstash-style "grok" patterns:
  [inputs.logparser.grok]
    ## This is a list of patterns to check the given log file(s) for.
    ## Note that adding patterns here increases processing time. The most
    ## efficient configuration is to have one pattern per logparser.
    ## Other common built-in patterns are:
    ##   %{COMMON_LOG_FORMAT}   (plain apache & nginx access logs)
    ##   %{COMBINED_LOG_FORMAT} (access logs + referrer & agent)
    patterns = ["<span style=\"font-weight: bold\">%{GREEDYDATA:APSMonitor:string}:</span><span style=\"font-weight: bold; color
: %{GREEDYDATA:Colour:string};\">%{GREEDYDATA:ResultCode:string}</span>"]
    #patterns = ["<span style=\"font-weight: bold\">%{GREEDYDATA:APSMonitor:string}"]

    ## Name of the outputted measurement name.
    measurement = "aps_diagnostics_monitor"

    ## Full path(s) to custom pattern files.
    #custom_pattern_files = []

    ## Custom patterns can also be defined here. Put one pattern per line.
    #custom_patterns = '''
    #'''

# OUTPUTS
[outputs]
[[outputs.influxdb]]
    # The full HTTP endpoint URL for your InfluxDB instance
    url = "http://<infludb server name>:8086" # EDIT THIS LINE
    # The target database for metrics. This database must already exist
    database = "<database name>" # required.
    skip_database_creation = true
    # retention_policy = "<retention policy name>"
    fielddrop = ["Colour"]
```

### System info:
Telegraf Version: 1.14.1
Operation system: RedHat Linux 7

### Steps to reproduce:
1) Place monitor.log file in the directory listed in the input.

2) Start Telegraf

### Expected behavior:

1) Each time I add a line to the monitor.log file, I expect Telegraf to send a measurement for each line.

### Actual behavior:

1) Each time I add a number of lines, Telegraf.log states that It sends a number of measurements equal to the number of lines in the file, which is what I would expect.

2) In Influxdb, only the measurement that corresponds to the last line is present. 

3) So if I have 10 lines in the files, Telegraf.log shows that 10 measurements are sent, but the measurement corresponding to the 10th line shows up in Influxdb.

4) If I add 5 lines to the file, telegraf.log shows that 15 measurements are sent, but the measurement corresponding to the 15ht line shows up in INfluxdb.

**Comment by tara-influxdata** [comment]
@ssoroka

**Comment by ssoroka** [comment]
It sounds like multiple metrics are getting collapsed into a single entry. This happens when they all have the same measurement name, tags, field names, and timestamp.

Can you provide an example of the metrics that are generated? you can get these with something like this:
```
[[outputs.file]]
  files = ["stdout", "output.txt"]
```

Also, please provide your [agent] settings.

**Comment by davidgarciaMontreal** [comment]
@gbodelle ive seen that behavior before. Can you provide log sample and the lines that you are adding? 
In my case, If each sample contained the same timestamp, then influxDB woudl only take the last one.

**Comment by gbodelle** [comment]
Hello Steven,

Thank you for your reply.  I ran the following scenario…  Please find the information below, I hope that it will help.


  1.  The original log file is:

<span style="font-weight: bold">MDA1 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA2 Files:</span><span style="font-weight: bold; color: green;">OK</span>

The output file is:

aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA2 Files",Colour="green",ResultCode="OK" 1595542489919444321
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log Colour="green",ResultCode="OK",APSMonitor="MDA1 Files" 1595542489919407530


  1.  I then added a line to the log file:

<span style="font-weight: bold">MDA1 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA2 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA3 Files:</span><span style="font-weight: bold; color: green;">OK</span>

The output file is:

aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA3 Files",Colour="green",ResultCode="OK" 1595542578661683384
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log Colour="green",ResultCode="OK",APSMonitor="MDA2 Files" 1595542578661650066
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA1 Files",Colour="green",ResultCode="OK" 1595542578661616706


  1.  I added two lines to the log file:

<span style="font-weight: bold">MDA1 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA2 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA3 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA4 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA5 Files:</span><span style="font-weight: bold; color: green;">OK</span>

The output file is:

aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA5 Files",Colour="green",ResultCode="OK" 1595542692018733592
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA4 Files",Colour="green",ResultCode="OK" 1595542692018693609
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log Colour="green",ResultCode="OK",APSMonitor="MDA3 Files" 1595542692018674238
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA2 Files",Colour="green",ResultCode="OK" 1595542692018630717
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA1 Files",Colour="green",ResultCode="OK" 1595542692018587847


In each case, we can see that the timestamp placed by Telegraf is different.  In each case, only the last line is added in Influxdb.

Any thought?  Am I thinking that the plugin works differently than it actually does?

Thank you for your help!

Sincerely,

Guillaume

**Comment by gbodelle** [comment]
Hello David,

Thank you for your reply.  I ran the following scenario…  Please find the information below, I hope that it will help.


  1.  The original log file is:

<span style="font-weight: bold">MDA1 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA2 Files:</span><span style="font-weight: bold; color: green;">OK</span>

The output file is:

aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA2 Files",Colour="green",ResultCode="OK" 1595542489919444321
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log Colour="green",ResultCode="OK",APSMonitor="MDA1 Files" 1595542489919407530


  1.  I then added a line to the log file:

<span style="font-weight: bold">MDA1 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA2 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA3 Files:</span><span style="font-weight: bold; color: green;">OK</span>

The output file is:

aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA3 Files",Colour="green",ResultCode="OK" 1595542578661683384
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log Colour="green",ResultCode="OK",APSMonitor="MDA2 Files" 1595542578661650066
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA1 Files",Colour="green",ResultCode="OK" 1595542578661616706


  1.  I added two lines to the log file:

<span style="font-weight: bold">MDA1 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA2 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA3 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA4 Files:</span><span style="font-weight: bold; color: green;">OK</span>
<span style="font-weight: bold">MDA5 Files:</span><span style="font-weight: bold; color: green;">OK</span>

The output file is:

aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA5 Files",Colour="green",ResultCode="OK" 1595542692018733592
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA4 Files",Colour="green",ResultCode="OK" 1595542692018693609
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log Colour="green",ResultCode="OK",APSMonitor="MDA3 Files" 1595542692018674238
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA2 Files",Colour="green",ResultCode="OK" 1595542692018630717
aps_diagnostics_monitor,host=<server>,path=diagnostics_monitor.log APSMonitor="MDA1 Files",Colour="green",ResultCode="OK" 1595542692018587847


In each case, we can see that the timestamp placed by Telegraf is different.  In each case, only the last line is added in Influxdb.

Any thought?  Am I thinking that the plugin works differently than it actually does?

Thank you for your help!

Sincerely,

Guillaume

**Comment by ssoroka** [comment]
@gbodelle, what are you using for `[agent]` settings? I'm specifically interested in the precision setting.

**Comment by gbodelle** [comment]
Hello,

It seems that the issue is with Chronograf.  I have yet to understand why it only displays the last measurement in the file.  I have tried to see if it is a display option.  However, when I copy the query from the Chronograf panel, and run it from the Influxdb command line, all of the expected measurements are displayed, in the expected format.  At this point, I am able to move forward and build graphs as I need.

I do not understand why Chronograf and Influx would show different results for the same query.  I have tried to investigate the possible options within Chronograf, but no luck.

Since I am able to move forward, I think that we can close the ticket.  Your help was very important, especially since I originally was using the logparse input, which I was told by support that it was deprecated.  So it was important that I switch to the tail input.

Again, thank you for your help!
